src/features/subscriptions/components/PricingContent/PricingContent.tsx

```diff
@@ -38,6 +40,29 @@ export function PricingContent() {
 
   const onUnauthorizedUserClick = useCallback(() => goTo('/profile'), []);
 
+  const onNewSubscriptionApproved = useCallback(() => {
+    showModal(SubscriptionSuccessModal).then(() => location.reload());
+  }, []);
+
+  useEffect(() => {
+    // after the page is loaded, set toggle to monthly/annually depending on current subscription
+    for (const plan of config.plans) {
+      if (currentSubscription?.id === plan.id) {
+        for (const cycle of plan.billingCycles) {
+          if (
+            cycle.billingMethods.find(
+              (billingMethod) =>
+                billingMethod.billingPlanId === currentSubscription.billingPlanId,
+            )
+          ) {
+            setCurrentBillingCycleID(cycle.id);
+            return;
+          }
+        }
```

Comment:
Maybe its better to het rid of some nesting here with something like this

```
    const currentPlan = config.plans.find((plan) => plan.id === currentSubscription?.id);
    if(!currentPlan) return;
    
    //rest of the code
```

Reply:
Fixed